python/pyarrow/src/arrow/python/platform.h

```diff
@@ -24,7 +24,9 @@
 // to mean Py_ssize_t (defining this to suppress deprecation warning)
 #define PY_SSIZE_T_CLEAN
 
-#include <Python.h>  // IWYU pragma: export
+// clang-format off: Keep pythoncapi_compat include first
+#include "arrow/python/vendored/pythoncapi_compat.h"  // IWYU pragma: export
```

Comment:
I wonder if it's a good idea to do this from a public header file. Third-party projects might then include their own vendored copy of a different version of `pythoncapi_compat.h` but then end up with the PyArrow version of that file?

Reply:
That's a good point. I'll change it so that `pythoncapi_compat.h` only gets included in source files.

Reply:
Done in https://github.com/apache/arrow/pull/43540/commits/174a1efda006d4b3ee7d99d7112aaa1b6088fc02.